### PR TITLE
Restructure Lesson 3: Remove WebApp, Smaller Modules, Integrate Exercises…

### LINKED ISSUE #64: Update Lesson 3: Remove web interface walkthrough, focus entirely on CLI
This lesson currently opens with a long, passive walkthrough of the Sensu web interface, then moves to a set of exercises built around the CLI/shell environment, which the remainder of the lessons also use. Instead, remove the web interface portion and focus only on the CLI content, moving the web interface walkthrough to a supplemental / later lesson (numbering TBD). 

Quantitative Metric Goals (as determined by this [Flesch-Kincaid calculator](https://goodcalculators.com/flesch-kincaid-calculator/)):

Flesh-Kincaid Grade Level: 8-10
Flesch Reading Ease Score: 40-60
Average Words per Sentence: 4-7
Average Syllables per Word: 1-2
Sentences: 75-100
Words: 375-500

### LINKED ISSUE #65: Update Lesson 3: reorganize into smaller units, interleave exercises w/ text
Currently the lesson is structured to have a lengthy section of informational text up-front, then a series of lessons at the end. Rather, interleave the instructional text and lessons, so that hands-on exercises happen immediately after covering the related information. 

Outline should be: 

- Title: **Lesson 3: Using the `sensuctl` CLI** 
- Goals: Explicitly describe Target Audience, Pre-Requisites, Intents.
- Exercise 1: Install and Configure the Sensu CLI 
- Exercise 2: Create an API Key
- Exercise 3: List Sensu Resources
- Exercise 4: View Resources and Event Details
- Discussion: Sensu is API-based, CLI is a client, Web app is also a client. Namespaces, note about output options, note about --help, etc.  
- Learn More: Links to more reading 
- Next Steps: Workshop Navigation (standard links: feedback, next step, prev step)

---

NOTE: This is part of an effort to standardize the structure of all the lessons to look something like this (with the associated quantitative lexicographical limits):

Lesson Structure (500-1000 words):
- Task Based Titles/Structure: “Creating a new Handler” (5-7 words)
- Goals: Explicitly describe Target Audience, Pre-Requisites, Intents. (50-100 words)
- Exercises: One more more exercise (see _Exercise Structure_ below)
- Discussion: Move any edge cases, extra explanation, caveats, related features, advanced uses cases, etc to this section. (50-100 words)
- Learn More: Links to more reading (3-7 links)
- Next Steps: Workshop Navigation (standard links: feedback, next step, prev step)

Exercise Structure:
- Task: short description of the task (25-50 words)
- Solution: short description of how we’ll solve the problem/complete the task (25-50 words)
- Steps: Numbered steps 1-5, what to do. (50-100 words per step)

Sentences should be 5-7 words.
Paragraphs should be 2-5 sentences long (10-35 words).
Sections should be ~50-100 words long (10-20 sentences, or ~3-5 paragraphs).
Lessons should be 500-1000 words long (above structure is within those bounds)
Lessons should have 3-5 exercises
Exercises should have 3-5 steps
Exercises should be ~250 words long, using about 50 words per step.

Example: Lower end of range = 705 words.
- Title: 5 words
- Goals: 50 words
- Exercise 1: Task: 25 words
- Exercise 1: Solution: 25 words
- Exercise 1: Step 1: 50 words
- Exercise 1: Step 2: 50 words
- Exercise 1: Step 3: 50 words
- Exercise 2: Task: 25 words
- Exercise 2: Solution: 25 words
- Exercise 2: Step 1: 50 words
- Exercise 2: Step 2: 50 words
- Exercise 2: Step 3: 50 words
- Exercise 3: Task: 25 words
- Exercise 3: Solution: 25 words
- Exercise 3: Step 1: 50 words
- Exercise 3: Step 2: 50 words
- Exercise 3: Step 3: 50 words
- Discussion: 50 words
- Learn More: 3 links (n/a word count)
- Next Steps: feedback, previous, next (n/a word count)

Quantitative Metric Goals (as determined by this [Flesch-Kincaid calculator](https://goodcalculators.com/flesch-kincaid-calculator/)):
- Flesh-Kincaid Grade Level: 8-10
- Flesch Reading Ease Score: 40-60
- Average Words per Sentence: 4-7
- Average Syllables per Word: 1-2
- Sentences: 150-200
- Words: 750-1000

### PR DESCRIPTION
**Pull request by thoward**
Closes #64, Closes #65.

This PR restructures Lesson 3 to remove the section on the web app, have smaller modules, interleaves the exercises with the explanation, and uses the new uniform lesson structure.

I kept the majority of the original content, but removed the exercise on `sensuctl dump` mostly for brevity's sake and because it felt like a use case that was beyond "absolute beginner". 

**Quantitative Analysis Using F-K Calculator (code samples removed):**

* Flesh-Kincaid Grade Level: 8.3
* Flesch Reading Ease Score: 47.7
* Reading Level: College ( Difficult to read )
* Average Words per Sentence: 6.8
* Average Syllables per Word: 1.8
* Sentences: 322
* Words: 2181*

_*NOTE: The overall length is still longer than I'd prefer, but I think it's within an acceptable limit. Would be good to do a "speed run" on this to see how long it takes to get through it with the new format._

View the rendered Markdown here: https://github.com/sensu/sensu-go-workshop/blob/381e36b700cce86133f1045d26d96ce9b6bc0c4a/lessons/operator/03/README.md

### CODE REVIEW

**Comment by thoward** [comment]
Aside: I will likely take one more pass through this before merge to add more links. During review if you notice a spot that looks like a good place for a link, please mention it!

Also, the "Lesson X" for Sensu Web App walkthrough still needs to be made, but not what numberin to use. Would like to have a quick chat about options for that.

File: lessons/operator/03/README.md [hunk 1]
```diff
@@ -1,370 +1,212 @@
...
+- [Monitoring as Code](#monitoring-as-code)
+  - [EXERCISE 6: Exporting and Updating Resource Configurations](#exercise-6-exporting-and-updating-resource-configurations)
 - [Learn more](#learn-more)
 - [Next steps](#next-steps)
 
-## Web App
+## Goals
 
-The Sensu backend process includes a web app that provides a unified view of your observability data (events, entities, and silences) and monitoring configuration (checks, handlers, etc) with user-friendly tools to reduce alert fatigue.
+In this lesson we will introduce the `sensuctl` command-line tool.
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: We could link to the CLI reference docs from here...

https://docs.sensu.io/sensu-go/latest/sensuctl/

File: lessons/operator/03/README.md [hunk 2]
```diff
@@ -1,370 +1,212 @@
...
+You will learn how to install and configure the tool, then practice performing some essential operations.
```
Comment (calebhailey) [inline lessons/operator/03/README.md:21]: s/tool/CLI/ ??

File: lessons/operator/03/README.md [hunk 3]
```diff
@@ -1,370 +1,212 @@
...
+This lesson is intended for operators of Sensu and assumes you have [set up a local workshop environment][setup_workshop].
 
-The Sensu web app homepage is a dashboard that provides a high level overview of the overall health of the systems under management by Sensu.
-The dashboard includes a summary of active incidents, a breakdown of incidents by severity, information about entities under management, and a high level overview of the namespaces (tenants).
-This dashboard complies with Sensu RBAC, so individual users will see summary information relevant to their access.
+## The `sensuctl` Command Line Interface (CLI)
 
-![The Sensu dashboard](img/03-01-dashboard.png)
+The `sensuctl` tool, short for _Sensu Control_, gives you full control of your Sensu pipeline from a command-line environment. 
+You can use `sensuctl` interactively in a shell, or script it as part of an automated solution.
```
Comment (calebhailey) [inline lessons/operator/03/README.md:28]: Should we offer at least one brief example of what "automation" means? Perhaps a parenthetical example like "(e.g. from your CI/CD pipeline)"??

File: lessons/operator/03/README.md [hunk 4]
```diff
@@ -1,370 +1,212 @@
...
+You want to view and manage resources in Sensu from a command-line environment or in an automated script.
 
-![The namespace switcher modal](img/03-02-namespace-switcher.png)
+#### Solution
 
-_**Figure 2:** the namespace switcher._
+Install the `sensuctl` CLI tool.
 
-![The "Select Namespace" option in the navigation sidebar](img/03-03-select-namespace.png)
+First, we will configure some helpful environment variables, download the `sensuctl` archive, then unpack that to a standard location on your system.
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: is this sentence missing a "then" in the middle? 

s/download/then download/ ??

Comment (calebhailey) [inline lessons/operator/03/README.md]: I also usually refer to these "archives" as just the contents, i.e. "binaries". This might read even easier if we just say "download `sensuctl`, then install it" – fully eliminating the concepts "archive" and "unpack".

File: lessons/operator/03/README.md [hunk 5]
```diff
@@ -1,370 +1,212 @@
...
+#### Steps
 
-### Event list & event detail views
+1. **Configure Environment Variables.**
 
-The default view that will be opened when navigating to a namespace is the event list view.
-A filter is applied to this view by default to show only events with a "non-passing" status (i.e. `event.check.state != 0-`).
-The event list view provides a number of options for filtering and sorting the list by event status, related entities, and more.
-Select one or more events in the event list view to take bulk actions like resolving or silencing the event(s).
+   The [workshop repository] includes platform-specific files that export some environment variables which we will use throughout the workshop.
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: unfinished link? is "workshop repository" in brackets because you were going to link it?

File: lessons/operator/03/README.md [hunk 6]
```diff
@@ -1,370 +1,212 @@
...
+     -LiteralPath "${Env:UserProfile}\sensu-go_${Env:SENSU_VERSION}_windows_amd64.zip" `
+     -DestinationPath "${Env:UserProfile}\Sensu\bin"
+   ${Env:Path} += ";${Env:UserProfile}\Sensu\bin"
+   ```
 
-1. **Log in to the Sensu web app.**
+   > **NOTE:** On Unix-like systems `/usr/local/bin` is the [standard location for binaries shared between multiple users][fhs_usr_local_docs]. 
+   > On macOS, the default permissions for `/usr/local/bin` require the used of `sudo`. 
+   > However, any location that is on your `$PATH` will work (i.e. `~/bin`).
+   > On Windows systems we create a new location specificly for Sensu, then add it to the path.
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: s/specificly/specifically/

File: lessons/operator/03/README.md [hunk 7]
```diff
@@ -1,370 +1,212 @@
...
 
-   Please visit http://127.0.0.1:3000 and login with the default workshop admin username (`sensu`) and password (`sensu`).
+## Configuration Options
```
Comment (calebhailey) [inline lessons/operator/03/README.md:130]: In case these headings ever get rolled up into a higher-level TOC it might be good to say "Sensuctl Configuration Options" or "CLI Configuration Options" so it's clear these are CLI config options, not Sensu Go configuration options.

File: lessons/operator/03/README.md [hunk 8]
```diff
@@ -1,370 +1,212 @@
...
-  - [Inventory](#inventory)
-  - [Output formats](#output-formats)
-  - [Interactive and non-interactive modes](#interactive-and-non-interactive-modes)
-- [EXERCISE 2: install and configure `sensuctl`](#exercise-2-install-and-configure-sensuctl)
-- [EXERCISE 3: use the `sensuctl --help` command](#exercise-3-use-the-sensuctl---help-command)
-- [EXERCISE 4: inspect an event in JSON format](#exercise-4-inspect-an-event-in-json-format)
-- [EXERCISE 5: explore Sensu resources using `sensuctl`](#exercise-5-explore-sensu-resources-using-sensuctl)
-- [EXERCISE 6: create an API Key for personal use](#exercise-6-create-an-api-key-for-personal-use)
-- [EXERCISE 7: export Sensu resources using `sensuctl dump`](#exercise-7-export-sensu-resources-using-sensuctl-dump)
+# Lesson 3: Using the Sensu CLI
```
Comment (calebhailey) [inline lessons/operator/03/README.md:1]: In the interest of introducing the simplest concepts first, I think we should bump the API Key content down. Perhaps in this order: 

1. Sensuctl CLI (i.e. the intro/overview/install section)
2. Configuration options 
3. Listing and viewing resources 
4. Inventory management 
5. Monitoring as Code (i.e. create -f)
6. API Keys

Thoughts?

Comment (calebhailey) [reply lessons/operator/03/README.md:1]: I do see how the "API Keys" section follows after the config section, so I'll defer to you on what you think is best here.

File: lessons/operator/03/README.md [hunk 9]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
+Many of the exercises in this workshop will require an API key, so let's create one now.
 
-Sensuctl includes a `--help` flag for getting help with every command and subcommand.
+### EXERCISE 3: Create an API Key
 
-Try running some of the following commands:
+#### Scenario
 
-1. **See all available `sensuctl` commands and global flags:**
+You want to use tools like `curl` and `sensuctl` to create and manage resources, but you do not want to use passwords.
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: Can we "active voice" here? Instead of saying "you don't want to use usernames and passwords" could we say "you prefer to use revoke-able API Keys instead of usernames and passwords"?

File: lessons/operator/03/README.md [hunk 10]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
-   sensuctl entity info learn.sensu.io
-   ```
+One of the common uses of `sensuctl` is to list and view resources like namespaces, users, and entities. 
 
-1. **Try exploring some other resources.**
+### EXERCISE 4: List Sensu Resources
+#### Scenario
 
-   _NOTE: don't forget to use `--help`; for example, `sensuctl --help` will output a list of "management commands" which are effectively API resources that are accessible via `sensuctl`._
+You want to explore the resources available in Sensu.
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: We should say "explore the resources under management by Sensu" (and you can optionally include "Sensu monitoring configuration" as another thing that can be explored in this manner). 

The term "available" will get us in trouble here, because in a fresh install we don't show you things that are "available" to do (e.g. a list of integrations that can be enabled). That's what our "Sensu Catalog" initiative aims to solve for in a future release.

File: lessons/operator/03/README.md [hunk 11]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
+   ```
+>**PROTIP:** Want to explore what `sensuctl` can do? Try using the  `--help` option to show a list of commands. This option is available on all commands and subcommands.
 
-1. **Save the API Key for use in a future exercise.**
+## Inventory Management
 
-   For the purposes of this workshop, we want to capture this API key (the
-   `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx` part of the output) for use in a future exercise.
-   You can either copy the output from the `sensuctl api-key grant` command manually, like this:
+One common use case for `sensuctl` is to manage a real-time "inventory of nodes".
```
Comment (calebhailey) [inline lessons/operator/03/README.md]: I would preface this by highlighting that one of the secondary features of Sensu as an infrastructure monitoring platform is that we provide a "real-time inventory" solution (or lightweight "endpoint management" solution). The fact that you can access this inventory data from `sensuctl` is a bonus (very cool when combined with `--label-selector` or `--field-selector` or plain old fashioned `sensuctl entity list --format json | jq .`.

Anyway, I'm also saying all this in case we can align this copy with the words we use on https://sensu.io/features (e.g. "endpoint management").

File: lessons/operator/03/README.md [hunk 12]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
+   OS:                     Workshop
+   Platform:               Sensu Go
+   Platform Family:        Training
+   Platform Version:       6.2.7
+   Auto-Deregistration:    true
+   Deregistration Handler: 
    ```
 
-1. **Export resources for a single namespace.**
+1. **Output Entity Information in JSON Format.**
```
Comment (calebhailey) [inline lessons/operator/03/README.md:440]: If we're going to show this example (which is great!), then we should _definitely_ hint that this is a super power of `sensuctl` when combined with other CLI tools such as `jq` (https://stedolan.github.io/jq/).

Comment (calebhailey) [reply lessons/operator/03/README.md:440]: I guess another way of saying this is "here's another opportunity to add a link"

File: lessons/operator/03/README.md [hunk 13]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
+   }
+   ```
+
+## Monitoring as Code
+
+The `sensuctl` monitoring-as-code workflow manages resource configurations using plain-text formats like YAML and JSON.
+
+Managing resources follows these basic steps:
+1. Resource configurations are defined in a YAML or JSON file.
+2. The `sensuctl create` command reads the resource configuration from the file, then pushes that configuration to the backend API where the resource is created.
```
Comment (calebhailey) [inline lessons/operator/03/README.md:518]: Should we s/sensuctl create/sensuctl create --file/ here (and explain that the more commonly used `-f` is shorthand for `--file`; or if we don't explain this should we use `--file` throughout the workshop)? You can technically use `sensuctl create` without `-f` or `--file` if you pipe some JSON or YAML to it, but that's a less common use case. 

I would describe the action performed by `sensuctl create --file` as instructing the Sensu CLI to "create [one or more resources] from this file", but in practice it does this by parsing the file contents (JSON or YAML), and _sending_ the resource data to the Sensu API which handles the actual "creation" (i.e. sensuctl only handles parsing, auth, and request; the Sensu API handles validation & creation). 

Probably too much detail to get into here, but I think adding `--file` here would help (even if just for verbosity).

File: lessons/operator/03/README.md [hunk 14]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
+       },
+       "created_by": "sensu"
+     },
+     "sensu_agent_version": ""
+   }
+   ```
+
+## Monitoring as Code
+
+The `sensuctl` monitoring-as-code workflow manages resource configurations using plain-text formats like YAML and JSON.
```
Comment (calebhailey) [inline lessons/operator/03/README.md:514]: I feel like there's a missed opportunity here to describe these files as "declarative" code/templates...

File: lessons/operator/03/README.md [hunk 15]
```diff
@@ -375,208 +217,446 @@ $ sensuctl configure -n \
...
+4. Later, the file can be modified and `sensuctl create` is used to update the resource.
+
+You can also use the `sensuctl dump` command to export a resource to a file, or add the `--format` option to a `sensuctl entity info` command to output the resource in a structured format. 
+
+These features together enable Sensu's _monitoring-as-code_ workflow.
+
+### EXERCISE 6: Exporting and Updating Resource Configurations
+#### Scenario
+
+You have an entity in a running Sensu environment but don't have a YAML file that describes it. You'd like to save the resource to a file so you can modify the configuration, and store that configuration to a code repository.
```
Comment (calebhailey) [inline lessons/operator/03/README.md:529]: Entities are probably not the right resource to use for this scenario (because in some cases, the Sensu API is not the source of truth for an entity – [the Sensu Agent is](https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-entities/entities/#manage-agent-entities-via-the-agent)). 

An alternative example could be that you are onboarding new team members who have created/modified checks using the Sensu web app (sidenote: this is just another API client) that you now wish to bring under management as code... then go through more-or-less the same workflow you have in the exercise. 

Sorry I didn't catch this sooner!

**Review by calebhailey — COMMENTED** [review]
Great job! I love the new structure. 

I left several comments with mostly minor nits and only one (!) typo.

File: lessons/operator/03/README.md (hunk 3 above)
Comment (thoward) [reply lessons/operator/03/README.md:28]: One of my goals with this re-write is to avoid the previous text's tendency towards parentheticals/digressions/i.e./e.g./notes/protips/etc.. This kind of meta text adds noise to the primary language and is only sometimes helpful for some readers. 

In this case, I think readers can infer this on their own, and maybe go into automation use cases in another part of the workshop if we want to focus on that.

File: lessons/operator/03/README.md (hunk 4 above)
Comment (thoward) [reply lessons/operator/03/README.md]: This is grammatically correct, though it's a niche case with some nuance. Here `then` is being used as a _coordinating conjunction_, similar to `and`. In this usage, it follows the same rules as `and` which means it's only required once, prefixing the last phrase. In fact, "I picked up the pen, wrote a note, and then signed my name." is grammatically equivalent to "I picked up the pen, wrote a note, then signed my name.". We can omit the `and` in this scenario because it's implied by the presence of `then`. That's because when `then` is used as a coordinating conjunction it has a stricter meaning than `and`. There is an implied order to `then` conjunctions, but no such order is required for `and`. In order to be both more precise and more concise, it's better to use the shorten form where possible.

Comment (thoward) [reply lessons/operator/03/README.md]: I agree. This is too much detail at this point. The detail is more useful in the actual exercise steps. Updating.

File: lessons/operator/03/README.md (hunk 5 above)
Comment (thoward) [reply lessons/operator/03/README.md]: Good catch!

File: lessons/operator/03/README.md (hunk 6 above)
Comment (thoward) [reply lessons/operator/03/README.md]: The typos never end.

File: lessons/operator/03/README.md (hunk 7 above)
Comment (thoward) [reply lessons/operator/03/README.md:130]: Good point, but I think I'll skip on this one. I think the context is sufficient to disambiguate that.

Also, one of the linguistic nits here is that any time I refer to `sensuctl` I want to do so w/ proper case and code formatting (e.g. in backticks). So that title would look like:

**`sensuctl` Configuration Options**

Which does not look good, because the first letter should be capitalized and using the code-symbol formatting in the heading font sizing just looks awkward, especially when it's the first word. 

I made sure to phrase things so that no sentences start with "<blah blah>. Sensuctl <blah blah>" or "<blah blah> . `sensuctl` <blah blah>" because of this.

File: lessons/operator/03/README.md (hunk 8 above)
Comment (thoward) [reply lessons/operator/03/README.md:1]: Yeah, I mostly put them together because functionally it makes sense to flow them that way, though I do agree that the API Key generation feels a bit premature here.

Comment (thoward) [reply lessons/operator/03/README.md:1]: Keeping as-is for now., but worth revisiting overall flow on a second pass, post-Illuminate.

File: lessons/operator/03/README.md (hunk 9 above)
Comment (thoward) [reply lessons/operator/03/README.md]: Good call. Also added a link to an article explaining the difference and why you might want to use API keys. 

https://medium.com/swlh/api-keys-whats-the-point-8f58d7966f9

File: lessons/operator/03/README.md (hunk 10 above)
Comment (thoward) [reply lessons/operator/03/README.md]: I don't love the phrase "under management by" because it's passively voiced and wordy. I went ahead and replaced it because I do see the distinction. 

I am not sure I completely agree that it's worth replacing the text for that though. "Available" here doesn't imply, to me, that it is a *complete* list of *possible* available things, but rather the things that are *currently available*/actionable/active/that Sensu is currently aware of/etc. Availability in a dist-sys context (what service nodes are available right now?) is a constantly changing thing, and implies that they are usable at the moment. Availability in the way you're using it here is more of a sale-mentality "We have these 5 models available from our supplier, though some might need to be special-ordered. What's in-stock today are these two models."...

Maybe we can brainstorm a phrase that means the same as my version of "available" without being too wordy or passive or implying other meanings.

File: lessons/operator/03/README.md (hunk 11 above)
Comment (thoward) [reply lessons/operator/03/README.md]: Makes sense! Although I will say that while I normally strongly encourage using the same terms everywhere, very consistently, there is an issue of voice. Many terms that primarily make sense in a marketing context don't sit well in a developer-focused/documentation context. They stand out as "buzzwordy" and lack definite meaning. See my commentary on this in the original review document. If possible we want to talk about things that are tangible and actionable and use the same terms that the tools use to describe them. There is no `sensuctl manage <endpoint>` command, so what does that translate to? Here, it translates to `sensuctl entities list`.. So rather than saying Sensu "manages endpoints" we could say it "lists entities" to be more rooted in the developer mindset. 

However, "listing entities" is also not well phrased to imply its use for endpoint management. This is a product issue that is larger than this individual document or workshop. Instead of calling them "entities" maybe we should call them "endpoints" in the code/events ... and instead of saying Sensu can "manage endpoints" we could say it can "show an inventory of endpoints" in the marketing copy.  

Then the command would look like `sensuctl endpoint list` and the marking copy would say `show an inventory of endpoints", which both use the same nouns, and *almost* the same verbs. The verb "manage" implies something well beyond inventorying to me.. Like creating new ones, starting/stopping existing one, changing endpoint configurations, etc. Sensu doesn't do all of that natively. So does the marketing copy really align with Sensu's capabilities? The gap there is revealed in situations like this. It's also unfortunately common for marketing speak to imply or suggest that something can do more than it actually does, but developer docs cannot afford the luxury of hyperbole. They must be grounded in the actual capabilities and terminology.

File: lessons/operator/03/README.md (hunk 12 above)
Comment (thoward) [reply lessons/operator/03/README.md:440]: Good call! Added a protip/example.

File: lessons/operator/03/README.md (hunk 14 above)
Comment (thoward) [reply lessons/operator/03/README.md:514]: I think most engineers understand that difference and don't need to be reminded of that explicitly.

File: lessons/operator/03/README.md (hunk 15 above)
Comment (thoward) [reply lessons/operator/03/README.md:529]: Yeah, I thought about this, but at this stage in the workshop, we really don't have much else to work with that makes sense here. I think this is relatively benign and it gently leads into the topic of subscriptions which we will be getting more into later. I think for now, let's leave this as-is, and maybe update the workshop environment to seed it with a resource that might be more interesting to use for this.

File: lessons/operator/03/README.md (hunk 13 above)
Comment (thoward) [reply lessons/operator/03/README.md:518]: IMO the `--file` portion of the command is not needed for the description at this point. Again, assuming our reader is a generally knowledgable user of UNIX-like command-line environments, it's understood that "everything is a file" including the STDIN stream, STDOUT stream, as well as any file you might use on disk. This is such an essential part of the UNIX command-line experience that mentioning it, or being explicit about it would just add noise. 

TBH if I had written this CLI tool I would not have required the user to explicitly use the `--file/-f` option and instead would have assumed the first argument to the subcommand create was a filename. If it was not present I would look on STDIN for input. If you want to create without a file or STDIN stream, using only `--whatever` parameters, then that would be the special case. Golden rule of product design is to make the most common case the easiest and least verbose/least steps. Putting a tax on the most common use case in the name of uniformity/consistency just alienates the user.

Comment (thoward) [reply lessons/operator/03/README.md:518]: It may make sense to use the verb "send" here vs "push". I think "send" might be more accurate anyway, and "push" is kind of a `git`-ism. Will update to that effect.

File: lessons/operator/03/README.md (hunk 2 above)
Comment (thoward) [reply lessons/operator/03/README.md:21]: I am a little flip-floppy on this one. I think I like using "CLI" when referring to Sensu the product and when describing `sensuctl`, but like using "tool" when referring to `sensuctl` the program.

That's because here CLI is not really a noun as much as it is an adjective in this usage. It is a "CLI tool/app" vs a "GUI tool/app" .. But it's confusing because CLI is actually a noun phrase "command-line interface" (interface is the noun), which itself uses TWO nouns combined as an adjective for "interface" ("command-line")... So when we say something is a CLI app, it's four nouns in a row, with the first three serving as adjectives. This is a terrible grammatical thing that is all too prevalent in tech. In the style-guides and tech-writing classes we used to teach at Twitter these were called "noun piles", and writers were cautioned to avoid them and were offered some strategies for how to undo them. 

Usually by leaving out some of the extra specifying words that don't realllly need to be there, or by adding some grammatical words into the mix to make the pile more parsable. In this case, we've avoided this by making an acronym of CLI, which helps somewhat, but also doesn't. 

"The command-line interface for Sensu is `sensuctl`". 
"`sensuctl` is a tool that provides a command-line interface to Sesnu".

Anyway -- this is not an easy thing to answer and that leads to a lot of inconsistent usage in the text, depending on which phrasing flows best for that particular sentence/phrase/title/etc.

Comment (thoward) [reply lessons/operator/03/README.md:21]: Leaving as-is for now, as I don't think there's an obviously better word choice here, and it's just as understandable as written.

File: lessons/operator/03/README.md (hunk 10 above)
Comment (calebhailey) [reply lessons/operator/03/README.md]: Opened an issue for follow up here: https://github.com/sensu/sensu-go-workshop/issues/76

File: lessons/operator/03/README.md (hunk 13 above)
Comment (calebhailey) [reply lessons/operator/03/README.md:518]: Not all of our users have Unix experience. We have a lot of Microsoft customers to whom many Unix/Linux concepts are quite foreign. Just FYI.
